src/corgi-tag-watcher.js

```diff
@@ -0,0 +1,82 @@
+import fetch from 'node-fetch'
+import sax from 'sax'
+import { ensureEnv } from './utils/ensureEnv.js'
+
+export default (app) => {
+  app.on('create', async (context) => {
+    const CORGI_URL = `https://${ensureEnv('CORGI_HOSTNAME')}/api/jobs/`
+    const SLACK_URL = `https://hooks.slack.com/services/${ensureEnv('CORGI_SLACK_SECRET')}`
+    // NOTE: if we miss webhooks look into persistence
+    const logger = app.log.child({ name: 'corgi-tag-watcher' })
+
+    logger.info('recieved webhook.')
+
+    // Do we need to filter events?
+    if (context.payload.ref_type !== 'tag') { return }
+    // get books in repo
+    const contentRequest = context.repo({ path: 'META-INF/books.xml' })
+    const contentMetadata = await context.octokit.repos.getContent(contentRequest)
+    const content = Buffer.from(contentMetadata.data.content, contentMetadata.data.encoding)
+
+    const repo = context.payload.repository.name
+
+    const books = []
+    const parser = sax.parser()
+    parser.onopentag = (node) => {
+      if (node.name === 'BOOK') {
+        books.push(node.attributes.SLUG)
+      }
+    }
+    parser.write(content).close()
+
+    // send job details to CORGI API
+    // make request per book + job type
+    let jobStatus = 'failed to queue'
+
+    try {
+      for (const slug of books) {
+        for (const jobType of [3, 4]) {
+          // 3: git-pdf
+          // 4: git-distribution-preview
+
+          logger.info(`collection_id: ${repo}/${slug}`)
+          logger.info(`job_type_id: ${jobType}`)
+
+          const payload = {
+            collection_id: `${repo}/${slug}`,
+            job_type_id: jobType,
+            status_id: 1,
+            version: `${context.payload.ref}`,
+            style: 'business-ethics' // TODO: add style to META-INF
```

Comment:
Oh -- just to make sure this doesn't get lost in the mix, do we have a story created to fill this gap in the `META-INF` schema?

Reply:
I'll create a card.

Reply:
Done.

Reply:
Just for future reference, I moved the issue to `cnx` (it looks like it was accidentally created in `webview`): https://github.com/openstax/cnx/issues/1615